theano/tensor/nnet/abstract_conv.py

```diff
@@ -1663,18 +1716,27 @@ def conv(self, img, kern, mode="valid", dilation=1, num_groups=1):
             raise ValueError(
                 'invalid dilation {}, expected {} values'.format(dilation,
                                                                  self.convdim))
+        if unshared and direction == "backprop weights":
+            if mode != "valid":
+                raise ValueError('conv mode for unshared backprop wrt weights must be "valid"')
+            # Do a transpose later to bring it to required shape
+            out_shape = (img.shape[0], kern.shape[0],
+                         kern.shape[2], kern.shape[3],
+                         img.shape[2] - kern.shape[2] + 1,
+                         img.shape[3] - kern.shape[3] + 1)
```

Comment:
There are some problems while doing that. The first two dimensions of img and topgrad are flipped before being passed to `conv`. So I can't use `get_conv_gradweights_shape`. And I can't use `get_conv_output_shape` since this is a special case (4d img and 4d kern -> 6d output).

Actually, `get_conv_output_shape` doesn't need an unshared parameter since the shape doesn't change. So it is complete as is.